extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,374 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import * as newProjectTool from '../tools/newProjectTool';
+import * as mssql from '../../../mssql';
+
+import { IconPathHelper } from '../common/iconHelper';
+import { cssStyles } from '../common/uiConstants';
+import { ImportDataModel } from '../models/api/import';
+import { Deferred } from '../common/promise';
+import { getConnectionName } from './utils';
+
+export class CreateProjectFromDatabaseDialog {
+	public dialog: azdata.window.Dialog;
+	public createProjectFromDatabaseTab: azdata.window.DialogTab;
+	public sourceConnectionTextBox: azdata.InputBoxComponent | undefined;
+	private selectConnectionButton: azdata.ButtonComponent | undefined;
+	public sourceDatabaseDropDown: azdata.DropDownComponent | undefined;
+	public projectNameTextBox: azdata.InputBoxComponent | undefined;
+	public projectLocationTextBox: azdata.InputBoxComponent | undefined;
+	public folderStructureDropDown: azdata.DropDownComponent | undefined;
+	private formBuilder: azdata.FormBuilder | undefined;
+	private connectionId: string | undefined;
+	private toDispose: vscode.Disposable[] = [];
+	private initDialogComplete!: Deferred<void>;
+	private initDialogPromise: Promise<void> = new Promise<void>((resolve, reject) => this.initDialogComplete = { resolve, reject });
+
+	public createProjectFromDatabaseCallback: ((model: ImportDataModel) => any) | undefined;
+
+	constructor(private profile: azdata.IConnectionProfile | undefined) {
```

Comment:
is profile = undefined a valid scenario at all? if not you should remove the undefined from its type.

Reply:
It is a valid scenario.

Reply:
With profile=undefined, the user can connect to a server (by clicking the SelectConnection Button).